tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -395,9 +399,343 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+		pods := createMultiplePods(ctx, client, pvclaims2d, true)
+		defer func() {
+			ginkgo.By("Delete pods")
+			deletePodsAndWaitForVolsToDetach(ctx, client, pods, true)
+		}()
+
+		rand.Seed(time.Now().Unix())
+		testdataFile := fmt.Sprintf("/tmp/testdata_%v_%v", time.Now().Unix(), rand.Intn(1000))
+		ginkgo.By(fmt.Sprintf("Creating a 100mb test data file %v", testdataFile))
+		op, err := exec.Command("dd", "if=/dev/urandom", fmt.Sprintf("of=%v", testdataFile),
+			"bs=1M", "count=100").Output()
+		fmt.Println(op)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			op, err = exec.Command("rm", "-f", testdataFile).Output()
+			fmt.Println(op)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expand pvcs while writing some data on them")
+		var wg sync.WaitGroup
+		currentPvcSize := pvcs[0].Spec.Resources.Requests[v1.ResourceStorage]
+		newSize := currentPvcSize.DeepCopy()
+		newSize.Add(resource.MustParse(diskSize))
+		fsSizes := []int64{}
+
+		for _, pod := range pods {
+			originalSizeInMb, err := getFSSizeMb(f, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			fsSizes = append(fsSizes, originalSizeInMb)
+		}
+		wg.Add(len(pods) * 2)
+		for i, pod := range pods {
+			go writeKnownData2PodInParallel(f, pod, testdataFile, &wg, fsSizes[i]-300)
+			go resize(client, pvcs[i], pvcs[i].Spec.Resources.Requests[v1.ResourceStorage], newSize, &wg)
+		}
+		wg.Wait()
+
+		ginkgo.By("Wait and verify the file system resize on pvcs")
+		for i := range pvcs {
+			framework.Logf("Waiting for file system resize to finish for pvc %v", pvcs[i].Name)
+			pvcs[i], err = waitForFSResize(pvcs[i], client)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			pvcConditions := pvcs[i].Status.Conditions
+			expectEqual(len(pvcConditions), 0, "pvc %v should not have status conditions", pvcs[i].Name)
+
+			var fsSize int64
+			framework.Logf("Verify filesystem size for mount point /mnt/volume1 for pod %v", pods[i].Name)
+			fsSize, err = getFSSizeMb(f, pods[i])
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			framework.Logf("File system size after expansion : %v", fsSize)
+			gomega.Expect(fsSize > fsSizes[i]).To(gomega.BeTrue(),
+				fmt.Sprintf(
+					"filesystem size %v is not > than before expansion %v for pvc %q",
+					fsSize, fsSizes[i], pvcs[i].Name))
+
+			framework.Logf("File system resize finished successfully")
+		}
+		ginkgo.By("Verify that the expanded CNS volumes are compliant and have correct policy id")
+		for i, pv := range pvs {
+			volumeID := pv.Spec.CSI.VolumeHandle
+			storagePolicyMatches, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyNames[i])
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(storagePolicyMatches).To(gomega.BeTrue(), "storage policy verification failed")
+			e2eVSphere.verifyVolumeCompliance(volumeID, true)
+		}
+
+		ginkgo.By("Verify the data on the PVCs match what was written in step 7")
+		for i, pod := range pods {
```

Comment:
We can combine this loop with the above loop for checking of compliance of volumes

Reply:
yes, but I'll keep it separate just so that it is more readable